modules/flake-parts/holochain-turn-server.nix

```diff
@@ -0,0 +1,184 @@
+{
+  # System independent arguments.
+  self,
+  lib,
+  ...
+}: {
+  perSystem = {
+    # Arguments specific to the `perSystem` context.
+    ...
+  }: {
+    # system specific outputs like, apps, checks, packages
+
+    # packages = ...
+  };
+  flake = {
+    # system independent outputs like nixosModules, nixosConfigurations, etc.
+
+    # nixosConfigurations.example-host = ...
+    overlays.coturn = _final: previous: {
+      coturn = previous.coturn.overrideAttrs (
+        _super: {
+          # coturn for NixOS needs to be built without libev_ok, otherwise acme-redirect won't work
+          LIBEV_OK = "0";
+          meta.platforms = lib.platforms.linux;
+        }
+      );
+    };
+
+    nixosModules.holochain-turn-server = {
+      config,
+      lib,
+      ...
+    }: let
+      cfg = config.services.holochain-turn-server;
+    in {
+      options.services.holochain-turn-server = {
+        enable = lib.mkEnableOption "holochain turn server";
+        url = lib.mkOption {
+          type = lib.types.str;
+        };
+        turn-cert-dir = lib.mkOption {
+          type = lib.types.str;
+          default = config.security.acme.certs.${cfg.url}.directory;
+        };
+        address = lib.mkOption {
+          type = lib.types.str;
+        };
+
+        nginx-http-port = lib.mkOption {
+          type = lib.types.int;
+          # skipping 81 because it's the default coturn alternative http port
+          default = 82;
+        };
+
+        coturn-min-port = lib.mkOption {
+          type = lib.types.int;
+          default = 20000;
+        };
+
+        coturn-max-port = lib.mkOption {
+          type = lib.types.int;
+          default = 65535; # which is default but here listing explicitly
+        };
+
+        verbose = lib.mkEnableOption "verbose logging";
+
+        acme-redirect = lib.mkOption {
+          type = lib.types.str;
+          default = "http://acme-${cfg.url}/.well-known/acme-challenge/";
+        };
+
+        username = lib.mkOption {
+          type = lib.types.str;
+          default = "test";
+        };
+
+        credential = lib.mkOption {
+          type = lib.types.str;
+          default = "test";
```

Comment:
according to @neonphog this isn't relevant for security. i don't have a strong opinion on this one.

Reply:
that was in the context of a discussion about the coturn admin cli. the creds are relevant for establishing turn connections

Reply:
> the creds are relevant for establishing turn connections

and since this is a public service we actually want this to be public. is that correct? or is there a point made for not defaulting to `test` here?

Reply:
I'd love it if we could find a way to pair users with creds, if nothing else so we could examine usage, but we haven't yet...